cmake/modules/CheckLinkerDynamicRelocationFlags.cmake

```diff
@@ -0,0 +1,11 @@
+include(CheckCXXCompilerFlag)
+
+function(check_linker_supports_flag flag_to_test var_name)
+  set(CMAKE_REQUIRED_FLAGS "${flag_to_test}")
+  check_cxx_compiler_flag("" LINKER_SUPPORTS_NOEXTERN_PROTECTED_DATA)
+  if(LINKER_SUPPORTS_NOEXTERN_PROTECTED_DATA)
```

Comment:
Hm, I understand that.
And I found CMake caches the result of `check_cxx_compiler_flag` *associated with the variable name*.
Because of that, if `"-Wl,-z,nocopyreloc"` failed, then `"-Wl,-z,noextern-protected-data"` fails without actual test.
To prevent that, `<var>` for every tests must have unique names.

For example:
```
function(check_linker_supports_flag flag_to_test var_name)                        
    set(CMAKE_REQUIRED_FLAGS "${flag_to_test}")                                   
    check_cxx_compiler_flag("" ${var_name})                                       
    if(${${var_name}})                                                            
        set("${var_name}" TRUE PARENT_SCOPE)                                      
    else()                                                                        
        set("${var_name}" FALSE PARENT_SCOPE)                                     
    endif()                                                                       
endfunction()                                                                     
```